src/main/java/net/dv8tion/jda/api/entities/User.java

```diff
@@ -225,4 +228,169 @@ default String getEffectiveAvatarUrl()
+        private final int offset;
+        private final int raw;
+        private final String name;
+
+        UserFlag(int offset, @Nonnull String name)
+        {
+            this.offset = offset;
+            this.raw = 1 << offset;
+            this.name = name;
+        }
+
+        /**
+         * The readable name as used in the Discord Client.
+         * 
+         * @return The readable name of this {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlag}.
+         */
+        @Nonnull
+        public String getName()
+        {
+            return this.name;
+        }
+
+        /**
+         * The binary offset of the flag.
+         * 
+         * @return The offset that represents this {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlag}.
+         */
+        public int getOffset()
+        {
+            return offset;
+        }
+
+        /**
+         * The value of this flag when viewed as raw value.
+         * <br>This is equivalent to: <code>1 {@literal <<} {@link #getOffset()}</code>
+         * 
+         * @return The raw value of this specific flag.
+         */
+        public int getRawValue()
+        {
+            return raw;
+        }
+
+        /**
+         * Gets the first {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlag} relating to the provided offset.
+         * <br>If there is no {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlag} that matches the provided offset,
+         * {@link net.dv8tion.jda.api.entities.User.UserFlag#UNKNOWN UserFlag.UNKNOWN} is returned.
+         * 
+         * @param  offset
+         *         The offset to match a {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlag} to.
+         *         
+         * @return {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlag} relating to the provided offset.
+         */
+        @Nonnull
+        public static UserFlag getFromOffset(int offset)
+        {
+            for (UserFlag flag : values())
+            {
+                if(flag.offset == offset)
+                    return flag;
+            }
+            return UNKNOWN;
+        }
+        
+        /**
+         * A set of all {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlags} that are specified by this raw int representation of
+         * flags.
+         * 
+         * @param  flags
+         *         The raw {@code int} representation if flags.
+         *         
+         * @return Possibly-empty EnumSet of {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlags}.
+         */
+        @Nonnull
+        public static EnumSet<UserFlag> getFlags(int flags)
+        {
+            final EnumSet<UserFlag> foundFlags = EnumSet.noneOf(UserFlag.class);
+            
+            if (flags == 0)
+                return foundFlags; //empty
+            
+            for (UserFlag flag : values())
+                if (flag != UNKNOWN && (flags & flag.raw) == flag.raw)
+                    foundFlags.add(flag);
```

Comment:
```suggestion
            for (UserFlag flag : values())
            {
                if (flag != UNKNOWN && (flags & flag.raw) == flag.raw)
                    foundFlags.add(flag);
            }
```

Didn't realize the convention for JDA was having braces on the loops